docs/samples/Microsoft.ML.Samples/Dynamic/IidSpikeDetectorTransform.cs

```diff
@@ -83,5 +87,84 @@ public static void IidSpikeDetectorTransform()
             // 0       5.00    0.50
             // 0       5.00    0.50
         }
+
+        public static void IidSpikeDetectorPrediction()
+        {
+            // Create a new ML context, for ML.NET operations. It can be used for exception tracking and logging, 
+            // as well as the source of randomness.
+            var ml = new MLContext();
+
+            // Generate sample series data with a spike
+            const int size = 10;
+            var data = new List<IidSpikeData>(size);
+            for (int i = 0; i < size / 2; i++)
+                data.Add(new IidSpikeData(5));
+            // This is a spike
+            data.Add(new IidSpikeData(10));
+            for (int i = 0; i < size / 2; i++)
+                data.Add(new IidSpikeData(5));
+
+            // Convert data to IDataView.
+            var dataView = ml.CreateStreamingDataView(data);
+
+            // Setup IidSpikeDetector arguments
+            string outputColumnName = "Prediction";
+            string inputColumnName = "Value";
+            var args = new IidSpikeDetector.Arguments()
+            {
+                Source = inputColumnName,
+                Name = outputColumnName,
+                Confidence = 95,                // The confidence for spike detection in the range [0, 100]
+                PvalueHistoryLength = size / 4  // The size of the sliding window for computing the p-value
+            };
+
+            // The transformed model.
+            ITransformer model = new IidSpikeEstimator(ml, args).Fit(dataView);
+
+            // Create a time series prediction engine from the model.
+            var engine = model.CreateTimeSeriesPredictionFunction<IidSpikeData, IidSpikePrediction>(ml);
+            for (int index = 0; index < 5; index++)
+            {
+                // Anomaly spike detection.
+                var prediction = engine.Predict(new IidSpikeData(5));
+                Console.WriteLine("{0}\t{1}\t{2:0.00}\t{3:0.00}", 5, prediction.Prediction[0],
+                    prediction.Prediction[1], prediction.Prediction[2]);
+            }
+
+            // Spike.
+            var spikePrediction = engine.Predict(new IidSpikeData(10));
+            Console.WriteLine("{0}\t{1}\t{2:0.00}\t{3:0.00}", 10, spikePrediction.Prediction[0],
+                spikePrediction.Prediction[1], spikePrediction.Prediction[2]);
+
+            // Checkpoint the model.
```

Comment:
>// Checkpoint the model. [](start = 12, length = 24)

why we doing this?

Reply:
We are checkpointing the model so that we can reload it with the updated saved from the predictions. Yep, I have addressed that as part of documentation.

---
In reply to: [237327645](https://github.com/dotnet/machinelearning/pull/1762#discussion_r237327645) [](ancestors = 237327645)

Reply:
Where is the documentation?  I wanted to see if I need to add anything for Martingale, P-Value, etc.

---
In reply to: [237702797](https://github.com/dotnet/machinelearning/pull/1762#discussion_r237702797) [](ancestors = 237702797,237327645)

Reply:
Its in this PR, check files under src/Microsoft.ML.TimeSeries